File: 900/1650B/main.cpp
#include<bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin>>t;

    while(t--) 
    {
        int l, r, a;
        cin>>l>>r>> a;

        // Maximum value can be achieved at r itself
        int ans = (r / a) + (r % a);

        // The maximum value could also occur right before the multiple of 'a'
        int m = (r / a) * a - 1;

        // Check if m is within the range [l, r]
        if (m >= l)
            ans = max(ans, (m / a) + (m % a));

        cout << ans << "\n";
    }

    return 0;
}

File: 900/1650B/main.java
import java.util.*;
public class main
{
	public static void main(String args[]) 
	{
    	Scanner sc = new Scanner(System.in);
    	int t = sc.nextInt();
 
    	while(t-- > 0)
    	{
    		int l = sc.nextInt();
    		int r = sc.nextInt();
    		int a = sc.nextInt();
 
    		int ans = 0, m = 0;
    		ans = (r/a) + (r%a);
    		m = r/a*a-1;

            // Check if m is within the range [l, r]
    		if(m >= l)
    			ans = Math.max(ans, (m/a)+(m%a));
 
    		System.out.println(ans);
		}
	}
}